core/src/main/java/org/mule/runtime/core/api/diagnostics/ProfilingDataProducer.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.runtime.core.api.diagnostics;
+
+import org.mule.api.annotation.Experimental;
+
+/**
+ * @param <T> the class that encapsulates the data for the profiling event context.
+ * @since 4.4
+ */
+@Experimental
+public interface ProfilingDataProducer<T extends ProfilingEventContext> {
```

Comment:
shouldn't this go into mule-api? o a new api module for using this functionality?

Reply:
Moved to mule-api. If this results in a more decoupling component after the experimental phase is finished, we'll move this to a different module. For the time being, the beans associated to the discovery service and profiling service will be  implemented and managed in the runtime.